### PR TITLE
XL/format: Fix a bug in checkDisksConsistency.

### PR DESCRIPTION
**Pull request by harshavardhana**
Current checkDisksConsistency was not exhaustive enough check fix it 
and add relevant unit tests as well.

### CODE REVIEW

File: format-config-v1.go [hunk 1]
```diff
@@ -224,10 +224,41 @@ func genericFormatCheck(formatConfigs []*formatConfigV1, sErrs []error) (err err
...
+	var orderIndexes []int
+	// Validate each for format.json for relevant uuid.
+	for _, formatConfig := range formatConfigs {
+		if formatConfig == nil {
+			continue
+		}
+		// Validate if UUID is present in JBOD.
+		uuidIndex := findDiskIndex(uuid, formatConfig.XL.JBOD)
+		if uuidIndex == -1 {
+			// UUID not found.
```
Comment (abperiasamy) [inline format-config-v1.go:243]: We should log here. We have all the context to this error.

File: format-config-v1.go [hunk 2]
```diff
@@ -224,10 +224,41 @@ func genericFormatCheck(formatConfigs []*formatConfigV1, sErrs []error) (err err
...
+		}
+		// Save the position of UUID present in JBOD.
+		orderIndexes = append(orderIndexes, uuidIndex+1)
+	}
+	// Once uuid is found, verify if the uuid
+	// present in same order across all format configs.
+	prevOrderIndex := orderIndexes[0]
+	for _, orderIndex := range orderIndexes {
+		if prevOrderIndex != orderIndex {
+			return false
```
Comment (abperiasamy) [inline format-config-v1.go:256]: Here as well.

File: format-config-v1_test.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	for i, testCase := range testCases {
+		err := checkFormatXL(testCase.formatConfigs)
+		if err != nil && testCase.shouldPass {
+			t.Errorf("Test %d: Expected to pass but failed with %s", i+1, err)
+		}
+		if err == nil && !testCase.shouldPass {
+			t.Errorf("Test %d: Expected to fail but passed instead", i+1)
+		}
+	}
+}
```
Comment (abperiasamy) [inline format-config-v1_test.go:193]: Instead of this style, it is better to write 1-1 unit tests. For example, isSavedUUIDInOrder will have isSavedUUIDInOrderTest. This test will only look out for variations in UUID order. This way, we will have a cleaner coverage for all functions. If we only test higher level functions, details may be lost. It is hard to remember all the tricky corner cases at higher level functions. Test functions for higher level functions will only test the glue code, how lower level functions are stitched together.

Comment (harshavardhana) [inline format-config-v1_test.go:193]: > Instead of this style, it is better to write 1-1 unit tests. For example, isSavedUUIDInOrder will have isSavedUUIDInOrderTest. This test will only look out for variations in UUID order. This way, we will have a cleaner coverage for all functions. If we only test higher level functions, details may be lost. It is hard to remember all the tricky corner cases at higher level functions. Test functions for higher level functions will only test the glue code, how lower level functions are stitched together.

Okay this make sense.
